plugins/jobtype/src/azkaban/jobtype/HadoopSecurePigWrapper.java

```diff
@@ -53,38 +52,23 @@
   }
 
   public static void main(final String[] args) throws Exception {
-    String propsFile = System.getenv(ProcessJob.JOB_PROP_ENV);
-    props = new Props(null, new File(propsFile));
-
+    Properties jobProps = HadoopSecureWrapperUtils.loadAzkabanProps();
+    props = new Props(null, jobProps);
     HadoopConfigurationInjector.injectResources(props);
 
-    final Configuration conf = new Configuration();
-
-    UserGroupInformation.setConfiguration(conf);
-    securityEnabled = UserGroupInformation.isSecurityEnabled();
-    pigLogFile = new File(System.getenv("PIG_LOG_FILE"));
-    if (!HadoopSecureWrapperUtils.shouldProxy(props.toProperties())) {
-      logger.info("Not proxying.");
-      runPigJob(args);
-      return;
-    }
-
-    UserGroupInformation proxyUser = null;
-    String userToProxy = props.getString("user.to.proxy");
-    if (securityEnabled) {
-      proxyUser = HadoopSecureWrapperUtils.createSecurityEnabledProxyUser(userToProxy, logger);
+    if (HadoopSecureWrapperUtils.shouldProxy(jobProps)) {
+      UserGroupInformation proxyUser =
+          HadoopSecureWrapperUtils.setupProxyUser(jobProps, logger);
+      proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
+        @Override
+        public Void run() throws Exception {
+          runPigJob(args);
+          return null;
+        }
+      });
     } else {
-      proxyUser = UserGroupInformation.createRemoteUser(userToProxy);
+      runPigJob(args);
```

Comment:
Just checked out setupProxyUser method in HadoopSecureWrapperUtils and it seems that we don't need if-else to check HadoopSecureWrapperUtils.shouldProxy(jobProps) as HadoopSecureWrapperUtils.setupProxyUser already takes care of that ?